### ISSUE TITLE
some kind of .ignore file?

### ISSUE THREAD

**Issue by mreinstein** [augmentable-dev/tickgit#32]
I'm running this project on a node/javascript project, and would like to ignore files in `node_modules/` `bundle.js`, etc. Apart from cobbling together an exact list of directories to watch, it would be great to be able to run `tickget status` in my working directory and ignore the parts I don't care about (vendor code, etc.)

Regardless, tickgit is a really cool concept!

**Comment by patrickdevivo** [comment]
Thanks for the support! I hope we can make something useful for a lot of developers.

Regarding your comment on ignoring vendor directories, we use a dependency called [`enry`](https://github.com/src-d/enry) to detect the language of files as they are searched. `enry` has a way of doing "vendor detection" which we use here: https://github.com/augmentable-dev/tickgit/blob/master/pkg/comments/comments.go#L88 to ignore (at least when looking for TODOs).

Did you run `tickgit status` or `tickgit todos` and see output from files in `node_modules`? I will add a test case and investigate!

**Comment by patrickdevivo** [comment]
I also agree that adding a way to ignore certain files or patterns (and do other types of config when the CLI is run) would be very useful - perhaps via a `.tickgit` or a `tickgit.yml`

**Comment by mreinstein** [comment]
On further inspection of the results you're right; it's_not_ including node_modules. 

> adding a way to ignore certain files or patterns (and do other types of config when the CLI is run) would be very useful

I wonder if it might make sense to just use the identical rules in `.gitignore` if it's present? I'm not totally sure if this is right, but maybe it makes sense. It stands to reason that anything one is ignoring from git would also be ignored in tickgit?

**Comment by patrickdevivo** [comment]
> I wonder if it might make sense to just use the identical rules in `.gitignore` if it's present? I'm not totally sure if this is right, just a thought. It stands to reason that anything one is ignoring from git would also be ignored by tickgit

That makes sense, though I could certainly see cases where a user might want to ignore certain paths that _are_ in the git repo, such as a non-standard dependency directory/area of third party code; or even just a folder/file they want to ignore adhoc (for instance, a test fixture or sample file)

**Comment by daurnimator** [comment]
For [zig](https://todos.tickgit.com/browse?repo=https://github.com/ziglang/zig) we have a vendored version of musl in `lib/libc/musl`: I'd like a way to exclude it.

**Comment by patrickdevivo** [comment]
Another request for this functionality: https://github.com/fyne-io/fyne/issues/683
